python_scripts/linear_models.py

```diff
@@ -159,8 +164,9 @@ def plot_data_and_model(
 
 # %% [markdown]
 # We can also see that we have a parameter `intercept_body_mass` in our model.
-# this parameter is the intercept of the curve when `x=0`. If the intercept is
-# null, then the curve will be passing by the origin:
+# this parameter is the value on the y-axis when `flipper_length=0` and it
+# is called the y-intercept. If `intercept_body_mass` is 0, the curve will
+# pass through the origin:
```

Comment:
Yes, we could have such a comment but instead of mentioning training data, we could mention possible data as we describe at first what was the possible range of the value for each feature. We could introduce the term extrapolation to explain that we don't want to go outside of this known statistic. This could remain quite high-level.

Reply:
see https://github.com/lucyleeow/scikit-learn-mooc/pull/2